drivers/net/ethernet/intel/idpf/idpf.h

```diff
@@ -244,51 +227,31 @@ struct idpf_adapter {
 	struct idpf_vector_lifo vector_stack;
 	irqreturn_t (*irq_mb_handler)(int irq, void *data);
 
-	u32 tx_timeout_count;
 	struct idpf_avail_queue_info avail_queues;
-	struct idpf_vport **vports;
-	struct net_device **netdevs;
-	struct virtchnl2_create_vport **vport_params_reqd;
-	struct virtchnl2_create_vport **vport_params_recvd;
-	u32 *vport_ids;
-
-	struct idpf_vport_config **vport_config;
 	u16 max_vports;
-	u16 num_alloc_vports;
-	u16 next_vport;
-
-	struct delayed_work init_task;
-	struct workqueue_struct *init_wq;
+	u16 default_vports;
 	struct delayed_work serv_task;
 	struct workqueue_struct *serv_wq;
 	struct delayed_work mbx_task;
 	struct workqueue_struct *mbx_wq;
 	struct delayed_work vc_event_task;
 	struct workqueue_struct *vc_event_wq;
-	struct delayed_work stats_task;
-	struct workqueue_struct *stats_wq;
+	struct delayed_work idc_eth_init_task;
+	struct workqueue_struct *idc_eth_init_wq;
 	struct virtchnl2_get_capabilities caps;
 	struct idpf_vc_xn_manager *vcxn_mngr;
 
 	struct idpf_dev_ops dev_ops;
 	int num_vfs;
 	bool crc_enable;
-	bool req_tx_splitq;
-	bool req_rx_splitq;
 
-	struct mutex vport_ctrl_lock;
+	struct mutex reset_lock;
```

Comment:
The function of vport_ctrl_lock was to protect instances where the vport data might be lost. For example if ethtool ndo arrives during the reset. From what I can see reset_lock is only taken in idpf_init_hard_reset() and not checked anywhere else, so I am not sure how we are using it to protect access during reset.

Reply:
The reset information is propagated to the aux drivers instance via event interface from PCI - IDC- eth. And it uses those flags for protection now.

Reply:
The RESET_IN_PROGRESS flag is not sufficient though, since it does not guarantee that vport data will be in good state. The use of flags was specifically rejected by the maintainer - the addition of vport_ctrl_lock was the last change that got us in. So this code aside from not actually protecting anything (just test with any ndo during reset) is not in a format that will be accepted by netdev. We need to figure out some mechanism for robust locking between the instances.

Reply:
I am trying to get over this code again to see if I can reapply the similar lock. I will update back on it.

Reply:
I think I addressed this issue finally. The code changes are in parity with the previous code now. The similar locking mechanism has been applied to the aux driver now.

Reply:
I hope we are okay with the changes you requested. Please let me know if we still have concern about it. I am marking it as resolved.